src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.CachedDocumentSearch.cs

```diff
@@ -120,84 +118,56 @@ private static void ClearCachedData()
 
         using var _1 = GetPooledHashSet(priorityDocumentKeys, out var priorityDocumentKeysSet);
 
-        // Sort the groups into a high pri group (projects that contain a high-pri doc), and low pri groups (those
-        // that don't).
+        // Sort the groups into a high pri group (projects that contain a high-pri doc), and low pri groups (those that
+        // don't), and process in that order.
         using var _2 = GetPooledHashSet(groups.Where(g => g.Any(priorityDocumentKeysSet.Contains)), out var highPriorityGroups);
         using var _3 = GetPooledHashSet(groups.Where(g => !highPriorityGroups.Contains(g)), out var lowPriorityGroups);
 
-        await ProcessProjectGroupsAsync(highPriorityGroups).ConfigureAwait(false);
-        await ProcessProjectGroupsAsync(lowPriorityGroups).ConfigureAwait(false);
-
+        await PerformParallelSearchAsync(
+            highPriorityGroups.Concat(lowPriorityGroups), ProcessSingleProjectGroupAsync, onItemsFound, cancellationToken).ConfigureAwait(false);
         return;
 
-        async Task ProcessProjectGroupsAsync(HashSet<IGrouping<ProjectKey, DocumentKey>> groups)
+        async ValueTask ProcessSingleProjectGroupAsync(
+            IGrouping<ProjectKey, DocumentKey> group,
+            Action<RoslynNavigateToItem> onItemFound,
+            CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-            using var _ = ArrayBuilder<Task>.GetInstance(out var tasks);
+            if (cancellationToken.IsCancellationRequested)
+                return;
 
-            foreach (var group in groups)
-                tasks.Add(ProcessProjectGroupAsync(group));
-
-            await Task.WhenAll(tasks).ConfigureAwait(false);
-        }
-
-        async Task ProcessProjectGroupAsync(IGrouping<ProjectKey, DocumentKey> group)
-        {
-            cancellationToken.ThrowIfCancellationRequested();
-            await Task.Yield().ConfigureAwait(false);
             var project = group.Key;
 
-            // Break the project into high-pri docs and low pri docs.
+            // Break the project into high-pri docs and low pri docs, and process in that order.
             using var _1 = GetPooledHashSet(group.Where(priorityDocumentKeysSet.Contains), out var highPriDocs);
```

Comment:
>using var _1 = GetPooledHashSet [](http://example.com/codeflow?start=12&length=31)

Similar question here

Reply:
this one can be simpler

Reply:
Why not use your Prioritize method here too?

Reply:
missed that case. happy to do in followup (unless htere is larger stuff found in this pass) :)